runtimes/runtimes/fqn/common/commonFqnWorkerPool.ts

```diff
@@ -0,0 +1,59 @@
+import { pool, Pool } from 'workerpool'
+import { DEFAULT_MAX_QUEUE_SIZE, DEFAULT_MAX_WORKERS, DEFAULT_TIMEOUT, FQN_WORKER_ID } from './defaults'
+import {
+    Cancellable,
+    ExtractFqnInput,
+    ExtractFqnResult,
+    IFqnWorkerPool,
+    Logging,
+    WorkerPoolConfig,
+} from '../../../server-interface'
+
+export class CommonFqnWorkerPool implements IFqnWorkerPool {
```

Comment:
why is a worker pool needed? it might be obvious now, but others working on this project in the future won't have the context that you have